framework/aster-frame/src/trap/handler.rs

```diff
@@ -102,3 +125,69 @@ cpu_local! {
 pub fn in_interrupt_context() -> bool {
     IN_INTERRUPT_CONTEXT.load(Ordering::Acquire)
 }
+
+fn kernel_page_fault_handler(f: &TrapFrame) {
+    // We only create mapping: `vaddr = paddr + PHYS_OFFSET` in kernel page fault handler.
+    let page_fault_vaddr = x86_64::registers::control::Cr2::read().as_u64();
+    debug_assert!(is_kernel_vaddr(page_fault_vaddr as usize));
+
+    // Check kernel region
+    // FIXME: The modification to the offset mapping of the kernel code and data should not permitted.
+    let kernel_region = MemoryRegion::kernel();
+    debug_assert!(
+        page_fault_vaddr < kernel_region.base() as u64
+            || page_fault_vaddr > (kernel_region.base() + kernel_region.len()) as u64
+    );
+
+    // Check error code and construct flags
+    let error_code = PageFaultErrorCode::from_bits_truncate(f.error_code);
+    debug!(
+        "Handling kernel page fault. Page fault address:{:x?}; Error code:{:?}",
+        page_fault_vaddr, error_code
+    );
+    debug_assert!(!error_code.contains(PageFaultErrorCode::USER));
+    debug_assert!(!error_code.contains(PageFaultErrorCode::INSTRUCTION));
+    let mut flags = PageTableFlags::empty()
+        .set_present(true)
+        .set_executable(false);
+    if error_code.contains(PageFaultErrorCode::WRITE) {
```

Comment:
Can directly use `let mut flags = PageTableFlags::PRESENT | PageTableFlags::EXECUTABLE`?

Reply:
Functions are used here to facilitate the multiple platforms support. When `PageTableFlags` in other platforms are different, using `PageTableFlags` directly will cause compilation errors. 

Besides, I think `PageFaultErrorCode` should be different on other platforms and should also follow the design of 'PageTableFlags'